### PR TITLE
Add Files IO guide

### PR DESCRIPTION
**Pull request by jonnius**
This is a guide explaining where an app can read and write files.

- [x] Show paths to Cache, Config and AppData
- [x] Explain what to use Cache, Config and AppData for
- [x] Show how to get these standard paths in C++
- [x] Show how to get these standard paths in QML
- [x] Add a note about app log

### CODE REVIEW

**Comment by jonnius** [comment]
Any change requests?

File: appdev/guides/writeable-dirs.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
+Writable directories
+====================
+
+App confinement is part of the Ubuntu Touch security concept. Data can be
+exchanged between apps only according to the AppArmor policies. This being said,
+apps can only read and write files that are located in one of three app 
+specific directories explained in this guide.
+
+.. Note::
+    The app can write debug messages into the app log file located at
```
Comment (UniversalSuperBox) [inline appdev/guides/writeable-dirs.rst]: Please use a whitespace checker to remove trailing whitepace from lines.

File: appdev/guides/writeable-dirs.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
+Writable directories
+====================
+
+App confinement is part of the Ubuntu Touch security concept. Data can be
```
Comment (UniversalSuperBox) [inline appdev/guides/writeable-dirs.rst]: As mentioned in [the Documentation contribute guide](https://docs.ubports.com/en/latest/contribute/documentation.html#line-length), there is no need to break lines for arbitrary length.

File: appdev/guides/writeable-dirs.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+apps can only read and write files that are located in one of three app 
+specific directories explained in this guide.
+
+.. Note::
+    The app can write debug messages into the app log file located at 
+    ``/home/phablet/.cache/upstart/application-click-<fullappname>_<appname>_<version>.log``.
+    To write into the log file, use the Qt debug functions.
+
+Config
+^^^^^^
```
Comment (UniversalSuperBox) [inline appdev/guides/writeable-dirs.rst]: RST is interesting in that many special characters are allowed for use in headers, and any character it finds will be used as the next heading level down.

By using `^^^` before `---`, the `^^^` became second-level headings and `---` third-level headings. This means that `StandardPaths in C++` and so on are subsections of` App data`. See [Headings in the documentation contribute section](https://docs.ubports.com/en/latest/contribute/documentation.html#headings) to find appropriate characters for each heading level.

File: appdev/guides/writeable-dirs.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+App confinement is part of the Ubuntu Touch security concept. Data can be
+exchanged between apps only according to the AppArmor policies. This being said,
+apps can only read and write files that are located in one of three app 
+specific directories explained in this guide.
+
+.. Note::
+    The app can write debug messages into the app log file located at 
+    ``/home/phablet/.cache/upstart/application-click-<fullappname>_<appname>_<version>.log``.
+    To write into the log file, use the Qt debug functions.
```
Comment (UniversalSuperBox) [inline appdev/guides/writeable-dirs.rst]: I'd recommend adding a link to the debug functions you're talking about

File: appdev/guides/writeable-dirs.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
+Writable directories
+====================
+
+App confinement is part of the Ubuntu Touch security concept. Data can be
+exchanged between apps only according to the AppArmor policies. This being said,
```
Comment (UniversalSuperBox) [inline appdev/guides/writeable-dirs.rst]: We might want to add a link to the ContentHub guides to let people know where to learn more about exchanging data.

File: appdev/guides/writeable-dirs.rst (hunk 1 above)
Comment (jonnius) [reply appdev/guides/writeable-dirs.rst]: Done

File: appdev/guides/writeable-dirs.rst (hunk 2 above)
Comment (jonnius) [reply appdev/guides/writeable-dirs.rst]: Done

File: appdev/guides/writeable-dirs.rst (hunk 3 above)
Comment (jonnius) [reply appdev/guides/writeable-dirs.rst]: Done

File: appdev/guides/writeable-dirs.rst (hunk 4 above)
Comment (jonnius) [reply appdev/guides/writeable-dirs.rst]: done

File: appdev/guides/writeable-dirs.rst (hunk 5 above)
Comment (jonnius) [reply appdev/guides/writeable-dirs.rst]: Done. I hope, I made the link correctly.

**Comment by jonnius** [comment]
Thanks for your feedback. Do you have more suggestions?

**Pull request merged**